src/Remapping/distributed_remapping.jl

```diff
@@ -770,17 +772,21 @@ function interpolate(remapper::Remapper, field::T) where {T <: Fields.Field}
 end
 
 function interpolate!(
-    dest::AbstractArray,
+    dest::Union{Nothing, <:AbstractArray},
     remapper::Remapper,
     field::T,
 ) where {T <: Fields.Field}
 
     axes(field) == remapper.space ||
         error("Field is defined on a different space than remapper")
 
-    size(dest) == size(remapper._interpolated_values) || error(
-        "Destination array is not compatible with remapper (size mismatch)",
-    )
+    if !isnothing(dest)
+        # !isnothing(dest) means that this is the root process, in this case, the size have
+        # to match
```

Comment:
What happens below if this is not the root process?

Reply:
Each MPI process has to run the code below. Each MPI process performs interpolation of its local points, then there is a MPI reduce! call to move everything to root and save it to `dest`.